parameterized/parameterized.py

```diff
@@ -40,6 +40,31 @@ def make_method(func, instance, type):
 
 _param = namedtuple("param", "args kwargs")
 
+
+def reapply_patches_if_need(func):
+
+    def dummy_wrapper(orgfunc):
+        @wraps(orgfunc)
+        def dummy_func(*args, **kwargs):
+            return orgfunc(*args, **kwargs)
+        return dummy_func
+
+    if hasattr(func, 'patchings'):
+        func = dummy_wrapper(func)
+        tmp_patchings = func.patchings
+        delattr(func, 'patchings')
+        for i, patch_obj in enumerate(tmp_patchings):
```

Comment:
The `tmp_patchings` is redundant; can just use `enumerate(func.patchings)`

Reply:
Here, actually we can't just use `enumerate(func.patchings)`, because we have to delete func.patching attribute before try to re-apply patches otherwise new patch object will be added to patchings array of original function, so somehow we have to keep original array to store patch objects and delete reference from `func` and then should re-apply all patches.